src/main/kotlin/com/bridgecrew/ui/vulnerabilitiesTree/CheckovToolWindowTree.kt

```diff
@@ -82,11 +89,53 @@ class CheckovToolWindowTree(val project: Project, val split: JBSplitter, private
             }
         }
 
+        expandsFromPathState(tree, globalState!!.expandedDescendants)
+
+        tree.addTreeExpansionListener(object : TreeExpansionListener {
+            override fun treeExpanded(event: TreeExpansionEvent?) {
+                globalState!!.expandedDescendants = getExpandedDescendants((event?.source as Tree))
+            }
+            override fun treeCollapsed(event: TreeExpansionEvent?) {
+                globalState!!.expandedDescendants = getExpandedDescendants((event?.source as Tree))
+            }
+        })
+
+
         resultsPanel.add(tree)
         isTreeEmpty = tree.isEmpty
         return resultsPanel
     }
 
+    private fun expandsFromPathState(tree:Tree, expandsTreePaths: List<TreePath>) {
+        expandsTreePaths.forEach {
+//            TODO: instead of searching the tree for each, expand while scanning
```

Comment:
is it still todo?

Reply:
kinda just a reminder if we want to improve the performance